### DISCUSSION TITLE
Need some help to fix the error when enabling emqx_exhook

### DISCUSSION THREAD

**Discussion by amalkasubasinghe** [emqx/emqx#9556] [Q&A]
Im using EMQX 4.4.11 and trying to add a hook following this https://github.com/emqx/emqx-extension-java-sdk

When I start the plugin it gives following error
```
022-12-15T09:14:32.374558+05:30 [error] [ExHook Svr] CALL emqx_exhook_v_1_hook_provider_client:on_provider_loaded(#{broker => #{datetime => "2022-12-15 09:14:32",sysdescr => "EMQ X Broker",uptime => "1 minutes, 1 seconds",version => "4.4.11"},meta => #{cluster_name => "emqxcl",node => <<"emqx@127.0.0.1">>,sysdescr => "EMQ X Broker",version => "4.4.11"}}, #{channel => "default",key_dispatch => <0.2007.0>,pool_size => 8,timeout => 5000}) error: econnrefused
2022-12-15T09:14:32.376430+05:30 [warning] Failed to reload exhook callback server "default", Reason: econnrefused
```
![Screenshot from 2022-12-15 09-18-47](https://user-images.githubusercontent.com/238594/207768167-148879c0-4a79-4c7e-ad9f-0614c1d6b66b.png)

Can someone share some tips to identify the root cause?

**Comment by lafirest** [comment]
@amalkasubasinghe 
This log showed your remote server (the grpc server) refused EMQX's connection. the reason may be:

1. you wrote a wrong server address
2. there is a firewall on the grpc server
3. the grpc server isn't listening
